src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -1058,6 +1131,8 @@ def gen_code(self, parent):
         '''
         Generate the f2pygen AST entries in the Schedule for this OpenMP do
         directive.
+        Note gen_code ignores the collapse clause but the generated code is
+        still valid.
```

Comment:
Presumably we'll fix this by removing `gen_code`. Please explain and maybe point to an issue.

Reply:
Done

Reply:
Thanks.